File: 35_Search_Insert_Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        flag=index=0
        for i in range(len(nums)):
            if nums[i]==target:
                flag=1
                return i
            
            if nums[i]>target and flag !=1:
                index=i
                break
                
        return index

File: 35_Search_Insert_Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        auto it=lower_bound(nums.begin(), nums.end(), target); 
        
        return it-nums.begin();
        
    }
};

Approch 2 using binary search:
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0, right = nums.size() - 1;

        while (left <= right)
        {
            int middle = left + (right - left) / 2;
            if (nums[middle] == target)
            {
                return middle;
            } 
			else if (nums[middle] < target)
            {
                left = middle + 1;
            }
            else
            {
                right = middle - 1;
            }
        }
		
        return left;
    }
};
